### PR TITLE
[flashlight] New Flashlight port

### PR DESCRIPTION
**Pull request by jacobkahn**
- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [ ] The versioning scheme in `vcpkg.json` matches what upstream says.
- [ ] The license declaration in `vcpkg.json` matches what upstream says.
- [ ] The installed as the "copyright" file matches what upstream says.
- [ ] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is in the new port's versions file.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Adela0814** [comment]
Pinging @jacobkahn  for response. Is work still being done for this PR?

**Comment by jacobkahn** [comment]
@Adela0814 - yes, this is still in progress. It’s contingent on other upstream changes in the project which are still in progress.

File: ports/flashlight/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "flashlight",
+  "version": "0.0.4",
```
Comment (MonicaLiu0311) [inline ports/flashlight/vcpkg.json:3]: I didn't find the version `0.0.4` upstream: https://github.com/flashlight/flashlight/tags.

**Comment by MonicaLiu0311** [comment]
At present, the version information of your commit is inconsistent with your local one:
```
##[error]Found the following errors:
##[error]versions/f-/flashlight.json(2,): error : while parsing versions for flashlight from version/f-/flashlight.json
0.0.4 is declared with 15ff66293d1444d14d09f00fd7e2444dece0a062, but the local port has a different SHA c9a8d69d96c808d825191cc939a6de8072d04e80.
```

`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version portName` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

**Comment by MonicaLiu0311** [comment]
Related upstream issues:
https://github.com/flashlight/flashlight/issues/1088
https://github.com/flashlight/flashlight/issues/1150

Related upstream PR:
https://github.com/flashlight/flashlight/pull/1137

@jacobkahn  You are an upstream author for this repository, and https://github.com/flashlight/flashlight/pull/1137 has been merged. We will continue to wait for your messages.

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Comment by jacobkahn** [comment]
@MonicaLiu0311 — Work is still being done. Waiting for the release of  two upstream dependencies to unblock further work here.

**Comment by jacobkahn** [comment]
It may be a few more weeks. If you want to close this PR, that's fine, I can reopen it when those are updated.

**Comment by MonicaLiu0311** [comment]
> It may be a few more weeks. If you want to close this PR, that's fine, I can reopen it when those are updated.

Okay, please notify me when the time comes to reopen it😄

**Pull request closed without merging**